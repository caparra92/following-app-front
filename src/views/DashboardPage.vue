<template>
 <ion-page>
    <ion-content>
        <ion-grid class="container">
            <ion-row>
                <ion-col>
                    <ion-avatar>
                        <img alt="Silhouette of a person's head" src="https://ionicframework.com/docs/img/demos/avatar.svg" />
                    </ion-avatar>
                </ion-col>
            </ion-row>
            <ion-row>
                <ion-col>
                    <p>Camilo</p>
                    <p>Parra</p>
                </ion-col>
            </ion-row>
            <div v-for="{ id, name } in categories" :key="id">
                <category-badge :activityType="name"/>
            </div>
        </ion-grid>
    </ion-content>
 </ion-page>
</template>

<script setup>
import { IonAvatar, IonPage, IonContent, IonIcon, IonGrid, IonRow, IonCol } from '@ionic/vue';
import CategoryBadge from '../components/CategoryBadge.vue';
import { useStore } from '../stores/store';
import { onMounted, ref } from 'vue';

const categories = ref([])

const store = useStore();

onMounted(async()=> {
 categories.value = await store.getCategories();
});

</script>

<style scoped>

.container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;
    height: 100vh;
    text-align: center;
  }

ion-avatar {
    align-self: center;
    margin: 0 auto;
}

ion-col p {
    font-size: 30px;
    margin: 0;
}

ion-col p:first-child {
    color: var(--ion-color-primary);
    font-weight: 600;
    margin-top: 30px;
}

ion-col p:last-child {
    color: var(--ion-color-medium);
    font-weight: 400;
    line-height: 1;
    margin-bottom: 30px;
}

</style>