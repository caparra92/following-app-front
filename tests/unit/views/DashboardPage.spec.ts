import { shallowMount, mount } from '@vue/test-utils';
import DashboardPage from '@/views/DashboardPage.vue';


describe('Dashboard tests', () => {
    test('Verify HTML content', () => {
        const wrapper = mount(DashboardPage);
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('Verify if HTML contains a text', () => {
        const wrapper = mount(DashboardPage);
        const p = wrapper.find('p');
        expect(p.text()).toBe('Camilo');
    });
});