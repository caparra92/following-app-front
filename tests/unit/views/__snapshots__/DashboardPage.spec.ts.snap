// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard tests Verify HTML content 1`] = `
<div class="ion-page">
  <ion-content class="">
    <ion-grid class="container">
      <ion-row>
        <ion-col>
          <ion-avatar class=""><img alt="Silhouette of a person's head" src="https://ionicframework.com/docs/img/demos/avatar.svg"></ion-avatar>
        </ion-col>
      </ion-row>
      <ion-row>
        <ion-col>
          <p>Camilo</p>
          <p>Parra</p>
        </ion-col>
      </ion-row>
      <ion-row class="">
        <ion-col class=""><button>
            <ion-icon></ion-icon>
          </button></ion-col>
        <ion-col class="">
          <p>Sports</p>
        </ion-col>
        <ion-col class=""><button>
            <ion-icon></ion-icon>
          </button></ion-col>
      </ion-row>
      <ion-row class="">
        <ion-col class=""><button>
            <ion-icon></ion-icon>
          </button></ion-col>
        <ion-col class="">
          <p>Reading</p>
        </ion-col>
        <ion-col class=""><button>
            <ion-icon></ion-icon>
          </button></ion-col>
      </ion-row>
      <ion-row class="">
        <ion-col class=""><button>
            <ion-icon></ion-icon>
          </button></ion-col>
        <ion-col class="">
          <p>Others</p>
        </ion-col>
        <ion-col class=""><button>
            <ion-icon></ion-icon>
          </button></ion-col>
      </ion-row>
    </ion-grid>
  </ion-content>
</div>
`;
